lib/loomio/loomio_handler.rb

```diff
@@ -0,0 +1,73 @@
+require 'httparty'
+
+class LoomioHandler
+  include HTTParty
+  @@base_uri = 'https://talk.theborderland.se/'
+  @@sessions_uri = 'api/v1/sessions'
+  @@comments_uri = 'api/v1/comments'
+  @@threads_uri = 'api/v1/discussions'
+
+  # login using username and password and set cookies
+  # for all the other requests within LoomioHandler
+  def initialize(username, password)
+    user = {user:{email:username,password:password}}
+    security_response = self.class.post(@@base_uri + @@sessions_uri, body: user)
+    puts(security_response.value) # raises error if the request failed
+
+    # parse and set security cookies based on the post call
+    @security_cookies = self.parse_set_cookie(security_response.headers['set-cookie'])
+    self.class.default_cookies.add_cookies(@security_cookies)
+
+    # set headers used for all requests
+    @headers = {
```

Comment:
NB that include HTTParty will allow you to set default headers with a class method:
```
class LoomioHandler
  include HTTParty
  base_uri 'https://talk.theborderland.se'
  headers 'Content-Type' => 'application/json'
```

This makes the api calls much simpler:
```
def new_thread(title, description = '')
  self.class.post('api/v1/discussions', body: {
    discussion: {
      title: title,
      description: description,
      group_id: 1259,
      private: false,
      forked_event_ids: [],
      document_ids: []
    }.to_json
  })
end
```